feat!: support `redirect.style.path` and `redirect.style.extension` (#618)
web-infra-dev/rslib: website/docs/en/config/lib/redirect.mdx

## redirect.style

### redirect.style.path

Whether to automatically redirect the import path when importing style files, the rules are the same as [redirect.js.path](/config/lib/redirect#redirectjspath).

- **Type:** `boolean`
- **Default:** `true`

- **Example:**

When importing normal style files:

```ts
import '@/foo.css'; // source code of './src/bar.ts' ↓
import './foo.css'; // expected output of './dist/bar.js'

import '@/foo.css'; // source code of './src/utils/index.ts' ↓
import '../foo.css'; // expected output of './dist/utils/index.js'
```

> When importing [CSS Modules](/zh/config/rsbuild/output#outputcssmodules) files:

Reviewer: ```suggestion
When importing [CSS Modules](/config/rsbuild/output#outputcssmodules) files:
```

```ts
import styles from '@/foo.css'; // source code of './src/bar.ts' ↓
import styles from './foo.css'; // expected output of './dist/bar.js'

import styles from '@/foo.css'; // source code of './src/utils/index.ts' ↓
import styles from '../foo.css'; // expected output of './dist/utils/index.js'
```

### redirect.style.extension

Whether to redirect style file extensions.

- **Type:** `boolean`
- **Default:** `true`

> When set to `true`, When importing a normal style file, the path will be rewritten to the `.css` file. When importing [CSS Modules](/zh/config/rsbuild/output#outputcssmodules), the path will be rewritten to the corresponding JavaScript output file.

Reviewer: ```suggestion
When set to `true`, if you import a normal style file, the path will be rewritten to the `.css` file and if you import [CSS Modules](/config/rsbuild/output#outputcssmodules), the path will be rewritten to the corresponding JavaScript output file.
```

## redirect.style

### redirect.style.path

Whether to automatically redirect the import path when importing style files, the rules are the same as [redirect.js.path](/config/lib/redirect#redirectjspath).

- **Type:** `boolean`
- **Default:** `true`

- **Example:**

When importing normal style files:

```ts
import '@/foo.css'; // source code of './src/bar.ts' ↓
import './foo.css'; // expected output of './dist/bar.js'

import '@/foo.css'; // source code of './src/utils/index.ts' ↓
import '../foo.css'; // expected output of './dist/utils/index.js'
```

> When importing [CSS Modules](/zh/config/rsbuild/output#outputcssmodules) files:

Reviewer: ```suggestion
When importing [CSS Modules](/config/rsbuild/output#outputcssmodules) files:
```

```ts
import styles from '@/foo.css'; // source code of './src/bar.ts' ↓
import styles from './foo.css'; // expected output of './dist/bar.js'

import styles from '@/foo.css'; // source code of './src/utils/index.ts' ↓
import styles from '../foo.css'; // expected output of './dist/utils/index.js'
```

### redirect.style.extension

Whether to redirect style file extensions.

- **Type:** `boolean`
- **Default:** `true`

> When set to `true`, When importing a normal style file, the path will be rewritten to the `.css` file. When importing [CSS Modules](/zh/config/rsbuild/output#outputcssmodules), the path will be rewritten to the corresponding JavaScript output file.

Reviewer: ```suggestion
When set to `true`, When importing a normal style file, the path will be rewritten to the `.css` file. When importing [CSS Modules](/config/rsbuild/output#outputcssmodules), the path will be rewritten to the corresponding JavaScript output file.
```
